ncbi/src/main/java/org/reactome/release/EuropePMC.java

```diff
@@ -0,0 +1,375 @@
+	 * @param reactomeVersion Release version of Reactome
+	 */
+	public static EuropePMC getInstance(String outputDir, int reactomeVersion) {
+		return new EuropePMC(outputDir, reactomeVersion);
+	}
+
+	private EuropePMC(String outputDir, int reactomeVersion) {
+		this.outputDir = outputDir;
+		this.reactomeVersion = reactomeVersion;
+	}
+
+	/**
+	 * Queries graph database for all human pathways with literature references
+	 * @param graphDBSession Neo4J Driver Session object for querying the graph database
+	 * @return Set of Europe PMC Link objects describing the pathway to literature reference annotations
+	 */
+	public static Set<EuropePMCLink> fetchEuropePMCLinks(Session graphDBSession) {
+		logger.info("Fetching Europe PMC Links from Reactome Pathways");
+
+		Set<EuropePMCLink> europePMCLinks = graphDBSession.run(
+			String.join(System.lineSeparator(),
+				"MATCH (p:Pathway)-[:literatureReference]->(lr:LiteratureReference)",
+				"WHERE p.speciesName = 'Homo sapiens' AND lr.pubMedIdentifier IS NOT NULL",
+				"RETURN DISTINCT p.displayName, p.stId, lr.pubMedIdentifier",
+				"ORDER BY p.displayName"
+			)
+		)
+		.stream()
+		.map(record ->
+			new EuropePMCLink(
+				record.get("p.displayName").asString(),
+				record.get("p.stId").asString(),
+				Integer.toString(record.get("lr.pubMedIdentifier").asInt())
+			)
+		).collect(Collectors.toCollection(LinkedHashSet::new));
+
+		logger.info("Finished fetching Europe PMC Links from Reactome Pathways");
+
+		return europePMCLinks;
+	}
+
+	/**
+	 * Writes Europe PMC profile and link files to pre-set output directory
+	 * @param graphDBSession Neo4J Driver Session object for querying the graph database
+	 * @throws IOException Thrown if creating or appending for either file fails
+	 */
+	public void writeEuropePMCFiles(Session graphDBSession) throws IOException {
+		logger.info("Writing Europe PMC files");
+
+		writeEuropePMCProfileFile();
+		writeEuropePMCLinkFile(graphDBSession);
+
+		logger.info("Finished writing Europe PMC files");
+	}
+
+	/**
+	 * Writes Europe PMC profile file to pre-set output directory.  Content describes Reactome, is static, and
+	 * is pre-defined.
+	 * @throws IOException Thrown if creating or appending for file fails
+	 */
+	private void writeEuropePMCProfileFile() throws IOException {
+		Path europePMCProfileFilePath = getEuropePMCProfileFilePath();
+
+		Files.deleteIfExists(europePMCProfileFilePath);
+		Files.createFile(europePMCProfileFilePath);
+
+		logger.info("Writing Europe PMC Profile file");
+
+		appendWithNewLine(getEuropePMCProfileXML(), europePMCProfileFilePath);
+
+		logger.info("Finished writing Europe PMC Profile file");
+	}
+
+	/**
+	 * Produces Europe PMC profile XML content
+	 * @return XML string describing Reactome as a data provider to Europe PMC
+	 */
+	private String getEuropePMCProfileXML() {
+		return String.join(System.lineSeparator(),
+			getXMLDeclaration(),
+			"<providers>",
+			indentString("<provider>", 1),
```

Comment:
For future reference, you can use JAXB to convert Java objects to XML. A simple example: https://www.mkyong.com/java/jaxb-hello-world-example/

Reply:
The JAXB looks good!  It might be a little too involved for this kind of XML, but I did find I can more flexibly create XML structures with org.w3c.dom package classes: https://docs.oracle.com/javase/8/docs/api/index.html?org/w3c/dom/package-summary.html

Reply:
The DOM packages also work well for creating XML.